### DISCUSSION TITLE
ai-bot-installer container image for storysakhi

### DISCUSSION THREAD

**Discussion by heungheung** [DJP-Digital-Jaaduii-Pitara/sakhi-api-service#107] [Q&A]
@kameshbhr not sure if this is the right place for the question or I should put it under the ai-bot-installer repo. Please let me know if I need to repost the question under a repo

basically, in the ai-bot-installer repo, for storysakhi 
https://github.com/Sunbird-AIAssistant/ai-bot-installer/blob/main/helmcharts/applications/charts/storysakhi/values.yaml#L7
the image points to
```
repository: story_sakhi_api_service
```
which does NOT contains any dockerhub user, organization name, nor any hosted container repository URL

Thus, deployment will always fail as no container can be pulled. 

If I take a look to the dockerhub user, where the other helm charts refer to,  
https://hub.docker.com/u/dubea  
I cannot find a correct container. Basically the corresponding "story" container 
https://hub.docker.com/r/dubea/sakhi-story-service
was actually build with this repo
https://github.com/Sunbird-AIAssistant/story-api-service
where it assumes the use of *OCI_** environment variable for the S3 storage bucket access
https://github.com/Sunbird-AIAssistant/story-api-service/blob/release-1.0.0/cloud_storage_oci.py

but the actual CM is using names BUCKET_ACCESS_KEY_ID, BUCKET_ENDPOINT_URL, BUCKET_NAME, BUCKET_REGION_NAME

Is there an updated container image for this storysakhi in AWS ai-bot-installer?  

TIA

**Comment by kameshbhr** [comment]
@amol-chitale Can you please check on this?

**Comment by AbhilashKD** [comment]
@heungheung 
AI installer is for the sakhi-api-service not for the story-sakhi so for sakhi-api-service here is the docker-image  :
dubea/sakhi-api-service-main-sakhi-api:main 

and its pushed in the repo as well link for the repo : 
https://github.com/Sunbird-AIAssistant/ai-bot-installer/blob/main/helmcharts/applications/charts/activitysakhi/values.yaml

**Reply by heungheung** [reply]
@AbhilashKD @kameshbhr In the readme file of the installer,
https://github.com/Sunbird-AIAssistant/ai-bot-installer
it is asking people to **Once ready with above details, run the installation script (install_on_aws.sh).**

while the install_on_aws.sh is doing
https://github.com/Sunbird-AIAssistant/ai-bot-installer/blob/main/install_on_aws.sh#L213
```
helm upgrade -i applications ./applications -n sb-ai-assistant --create-namespace -f global-values.yaml -f global-cloud-values.yaml
```
which will install/upgrade ALL charts under the applications folder

My 2 cents, if there is something not supported to be installed, it should be removed **OR** in the readme, we should explicitly state *which pod should be up and running* and *what can be ignored when people is doing kubectl get po -n sb-ai-assistant*
